app/code/Magento/InventoryCatalog/Test/Integration/InventoryCatalog/Model/Source/Validator/PreventDisableValidatorTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Test\Integration\InventoryCatalog\Model\Source\Validator;
+
+use Magento\Framework\Webapi\Exception;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\TestFramework\TestCase\WebapiAbstract;
+
+class PreventDisableValidatorTest extends WebapiAbstract
+{
+    /**#@+
+     * Service constants
+     */
+    const RESOURCE_PATH = '/V1/inventory/source';
+    const SERVICE_NAME = 'inventoryApiSourceRepositoryV1';
+    /**#@-*/
+
+    /**
+     * @var array
+     */
+    private $validData = [
+        SourceInterface::SOURCE_CODE => 'source-code-1',
+        SourceInterface::NAME => 'source-name-1',
+        SourceInterface::POSTCODE => 'source-postcode',
+        SourceInterface::COUNTRY_ID => 'US',
+    ];
+
+    /**
+     * @param string $field
+     * @param $value
+     * @param array $expectedErrorData
+     * @dataProvider failedValidationDataProvider
+     * @throws \Exception
+     */
+    public function testFailedValidationOnUpdate(string $field, $value, array $expectedErrorData)
+    {
+        $data = $this->validData;
+        $data[$field] = $value;
+
+        $sourceCode = 'default';
+        $serviceInfo = [
+            'rest' => [
+                'resourcePath' => self::RESOURCE_PATH . '/' . $sourceCode,
+                'httpMethod' => Request::HTTP_METHOD_PUT,
+            ],
+            'soap' => [
+                'service' => self::SERVICE_NAME,
+                'operation' => self::SERVICE_NAME . 'Save',
+            ],
+        ];
+        $this->webApiCall($serviceInfo, $data, $expectedErrorData);
+    }
+
+    /**
+     * SuppressWarnings was added due to a tests on different fail types and big size of data provider
+     *
+     * @return array
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     */
+    public function failedValidationDataProvider(): array
```

Comment:
According to class name `PreventDisableValidatorTest.php` there is only one case for testing
Why do we need dataprovider?
Need to remove dataprovider or rename test class

Reply:
I'll rename this test to `DisableDefaultSourceTest`. Is it ok?